Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm

```diff
@@ -1553,10 +1553,11 @@ static void addRedirectURL(WebHistoryItem *item, const String& url)
         auto element = adoptNS([[WebElementDictionary alloc] initWithHitTestResult:core(m_webFrame.get())->eventHandler().hitTestResultAtPoint(mouseEventData->absoluteLocation, hitType)]);
         [result setObject:element.get() forKey:WebActionElementKey];
 
-        if (mouseEventData->isTrusted)
-            [result setObject:@(mouseEventData->button) forKey:WebActionButtonKey];
-        else
-            [result setObject:@(WebCore::NoButton) forKey:WebActionButtonKey];
+        const auto toUnderlyingType = [](MouseButton button) {
+            return static_cast<std::underlying_type_t<MouseButton>>(button);
+        };
+        auto button = mouseEventData->isTrusted ? toUnderlyingType(mouseEventData->button) : toUnderlyingType(MouseButton::NoButton);
```

Comment:
This might actually be simpler as:

```suggestion
        auto button = static_cast<std::underlying_type_t<MouseButton>>(mouseEventData->isTrusted ? mouseEventData->button : MouseButton::NoButton);
```

(and then you don't need the lambda)

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/4922291bd8adbd2dfa8a9054b5ac748a657ca4da